### PR TITLE
[QNN] Support Dequantize to "float16" and Quantize to "uint16"

### PR DESCRIPTION
**Pull request by Johnson9009**
Current the Relay qnn.dequantize lack of the support to "float16" output, only can dequantize a quant value to "float32", this PR add the lack function, so as "uint16" of qnn.quantize.

### CODE REVIEW

**Review by leandron — APPROVED** [review]
LGTM, thanks @Johnson9009 for the PR and for fixing documentation issues along the way.

File: python/tvm/relay/qnn/op/qnn.py [hunk 1]
```diff
@@ -256,7 +256,7 @@ def simulated_quantize(data, output_scale, output_zero_point, axis=-1, out_dtype
     return _make.simulated_quantize(data, out_dtype, output_scale, output_zero_point, axis)
 
 
-def dequantize(data, input_scale, input_zero_point, axis=-1):
+def dequantize(data, input_scale, input_zero_point, axis=-1, out_dtype="float32"):
     r"""Dequantize op
     This operator takes quantized int8 and unit8 as input and produces
```
Comment (ibsidorenko) [inline python/tvm/relay/qnn/op/qnn.py:261]: Looks like this description is not correct anymore after all changes in input/output dtypes.

Comment (Johnson9009) [reply python/tvm/relay/qnn/op/qnn.py]: Done.

**Comment by Johnson9009** [comment]
@leandron  @ibsidorenko @junrushao @tqchen @Hzfengsy 
Is there anything need to change? If not, please help to check if we can merge it? Thanks.

**Pull request merged**